File: MyHashset.java
// Time Complexity :
// Space Complexity :
// Did this code successfully run on Leetcode :
// Any problem you faced while coding this :
class MyHashSet {
    int size = 1000000;
    LinkedList<Integer>[] userlist;

    private int get_index(int key) { // Getting the index for the key
        return key % size;
    }

    public MyHashSet() {
        userlist = new LinkedList[size];
    }

    public void add(int key) {
        int i = get_index(key);
        if (userlist[i] == null) {
            userlist[i] = new LinkedList<>();
        }
        if (userlist[i].indexOf(key) == -1)// adding the key because it's not present in the list
        {
            userlist[i].add(key);
        }
    }

    public void remove(int key) {
        int i = get_index(key);
        if (userlist[i] == null) {
            return;
        }
        if (userlist[i].indexOf(key) != -1) // removing the key because it is present in the list
        {
            userlist[i].remove(userlist[i].indexOf(key));
        }
    }

    public boolean contains(int key) {
        int i = get_index(key);
        // if the key is not present in the list returns false
        if (userlist[i] == null || userlist[i].indexOf(key) == -1) {
            return false;
        }
        //if the key is present in the list returns true
        return true;
    }
}
/**
 * Your MyHashSet object will be instantiated and called as such:
 * MyHashSet obj = new MyHashSet();
 * obj.add(key);
 * obj.remove(key);
 * boolean param_3 = obj.contains(key);
 */

File: myhashset.py
# Time Complexity : O(1)
# Space Complexity : O(N + K) for N buckets and K keys inserted into the hashmap
# Did this code successfully run on Leetcode : Yes
# Any problem you faced while coding this : Yes, it took me a few tries to write the 
# initialisation of the 2d array as a hashmap in Python correctly



class MyHashSet:

    def __init__(self):
        self.buckets = 1000
        self.bucketItems = 1000
        self.myhashset = [None] * self.buckets
    
    def bucket(self, key) -> int:
        return key % self.buckets
    
    def bucketItem(self, key) -> int:
        return key // self.bucketItems
    
    def add(self, key) -> None:
        bucket = self.bucket(key)
        bucketItem = self.bucketItem(key)
        if self.myhashset[bucket] is None:
            if bucket == 0:
                self.myhashset[bucket] = [False] * (self.bucketItems + 1)
            else:
                self.myhashset[bucket] = [False] * (self.bucketItems)
        self.myhashset[bucket][bucketItem] = True
    
    def remove(self, key) -> None:
        bucket = self.bucket(key)
        bucketItem = self.bucketItem(key)
        if self.myhashset[bucket] is None:
            return
        self.myhashset[bucket][bucketItem] = False
    
    def contains(self, key) -> bool:
        bucket = self.bucket(key)
        bucketItem = self.bucketItem(key)
        if self.myhashset[bucket] is None:
            return False
        return self.myhashset[bucket][bucketItem]


h1 = MyHashSet()
h1.add(51)
h1.add(61)
h1.remove(61)
print(h1.contains(51))